gammapy/catalog/hawc.py

```diff
@@ -0,0 +1,126 @@
+    _source_index_key = 'catalog_row_index'
+
+    def info(self, info='all'):
+        """Print info.
+
+        Parameters
+        ----------
+        info : {'all', 'basic', 'position', ' spectra'}
+            Comma separated list of options
+        """
+        ss = self.__str__(info=info)
+        print(ss)
+
+    def __str__(self, info='all'):
+        """Info string.
+
+        Parameters
+        ----------
+        info : {'all', 'basic', 'position', 'spectra} #TODO: call spectra different
+            Comma separated list of options
+        """
+
+        if info == 'all':
+            info = 'basic,position,spectra'
+
+        ss = ''
+        ops = info.split(',')
+        if 'basic' in ops:
+            ss += self._info_basic()
+        if 'position' in ops:
+            ss += self._info_position()
+        if 'spectra' in ops:
+            ss += self._info_spectra() #TODO: call spectra different
+        return ss
+
+    def _info_basic(self):
+        """Print basic info."""
+        d = self.data
+        ss = '\n*** Basic info ***\n\n'
+        ss += 'Catalog row index (zero-based) : {}\n'.format(d['catalog_row_index'])
+        ss += '{:<15s} : {}\n'.format('Source name:', d['source_name'])
+        return ss
+
+    def _info_position(self):
+        """Print position info."""
+        d = self.data
+        ss = '\n*** Position info ***\n\n'
+
+        ss += 'Measurement:\n'
+        ss += '{:20s} : {:.3f}\n'.format('RA', d['ra'])
+        ss += '{:20s} : {:.3f}\n'.format('DEC', d['dec'])
+        ss += '{:20s} : {:.3f}\n'.format('GLON', d['glon'])
+        ss += '{:20s} : {:.3f}\n'.format('GLAT', d['glat'])
+        ss += '{:20s} : {:.3f}\n'.format('Position error', d['pos_err'])
+
+        return ss
+
+    def _info_spectra(self):
+        """Print spectral info."""
+        d = self.data
+        ss = '\n*** Spectral info ***\n\n'
+
+        ss += 'Spectrum 1:\n'
+        ss += '{:20s} : {:.3} +- {:.3} {}\n'.format('Flux at 7 TeV', d['spec0_dnde'].value, \
+            d['spec0_dnde_err'].value, 'cm-2 s-1 TeV-1')
+        ss += '{:20s} : {:.3f} +- {:.3f}\n'.format('Spectral index', d['spec0_index'], d['spec0_index_err'])
+        ss += '{:20s} : {:1}\n\n'.format('Test radius', d['spec0_radius'])
+
+        if(np.isnan(d['spec1_dnde'])):
+            ss += 'No second spectrum available for this source'
+        else:
+            ss += 'Spectrum 2:\n'
+            ss += '{:20s} : {:.3} +- {:.3} {}\n'.format('Flux at 7 TeV', d['spec1_dnde'].value, \
+                d['spec1_dnde_err'].value, 'cm-2 s-1 TeV-1')
+            ss += '{:20s} : {:.3f} +- {:.3f}\n'.format('Spectral index', d['spec1_index'], d['spec1_index_err'])
+            ss += '{:20s} : {:1}'.format('Test radius', d['spec1_radius'])
+        return ss
+
+class SourceCatalog2HWC(SourceCatalog):
+    """ HAWC 2FHL catalog
+
+    One source is represented by `~gammapy.catalog.SourceCatalogObjectGammaCat`
+
+    See: TODO: Add here the reference_id?
```

Comment:
Should I add the webadress or the reference_id?

Reply:
Please add the link to ADS.

Reply:
Fixed